Physics3D/geometry/triangleMeshAVX.cpp

```diff
@@ -73,29 +73,30 @@ Vec3f TriangleMesh::furthestInDirectionAVX(const Vec3f& direction) const {
 
 	size_t offset = getOffset(vertexCount);
 	const float* xValues = this->vertices;
-	const float* yValues = this->vertices + offset;
-	const float* zValues = this->vertices + 2 * offset;
+	const float* yValues = this->vertices + 8;
+	const float* zValues = this->vertices + 16;
 
 	__m256 bestX = _mm256_load_ps(xValues);
 	__m256 bestY = _mm256_load_ps(yValues);
 	__m256 bestZ = _mm256_load_ps(zValues);
 
 	__m256 bestDot = _mm256_fmadd_ps(dz, bestZ, _mm256_fmadd_ps(dy, bestY, _mm256_mul_ps(dx, bestX)));
 
+	
 	for(size_t blockI = 1; blockI < (vertexCount + 7) / 8; blockI++) {
-		__m256i indices = _mm256_set1_epi32(int(blockI));
-
-		__m256 xVal = _mm256_load_ps(xValues + blockI * 8);
-		__m256 yVal = _mm256_load_ps(yValues + blockI * 8);
-		__m256 zVal = _mm256_load_ps(zValues + blockI * 8);
-
-		__m256 dot = _mm256_fmadd_ps(dz, zVal, _mm256_fmadd_ps(dy, yVal, _mm256_mul_ps(dx, xVal)));
-
-		__m256 whichAreMax = _mm256_cmp_ps(dot, bestDot, _CMP_GT_OQ); // Greater than, false if dot == NaN
-		bestDot = _mm256_blendv_ps(bestDot, dot, whichAreMax);
-		bestX = _mm256_blendv_ps(bestX, xVal, whichAreMax);
-		bestY = _mm256_blendv_ps(bestY, yVal, whichAreMax);
-		bestZ = _mm256_blendv_ps(bestZ, zVal, whichAreMax);
+	  __m256i indices = _mm256_set1_epi32(int(blockI));
+	  __m256 xVal = _mm256_load_ps(xValues + blockI * 24);
+	  __m256 yVal = _mm256_load_ps(yValues + blockI * 24);
+	  __m256 zVal = _mm256_load_ps(zValues + blockI * 24);
+	  
+	  __m256 dot = _mm256_fmadd_ps(dz, zVal, _mm256_fmadd_ps(dy, yVal, _mm256_mul_ps(dx, xVal))); 
+	  
+	  __m256 whichAreMax = _mm256_cmp_ps(dot, bestDot, _CMP_LT_OQ); // Greater than, false if dot == NaN
+	
+	  bestDot = _mm256_blendv_ps(dot, bestDot, whichAreMax);
+	  bestX = _mm256_blendv_ps(xVal, bestX, whichAreMax);
+	  bestY = _mm256_blendv_ps(yVal, bestY, whichAreMax);
+	  bestZ = _mm256_blendv_ps(zVal, bestZ, whichAreMax);
```

Comment:
Looks great! The code above with all commented out debug prints should look more like this.